src/quicktype-core/language/JavaScript.ts

```diff
@@ -83,7 +84,7 @@ export function nameStyle(
         legalizeName,
         upper ? firstUpperWordStyle : allLowerWordStyle,
         firstUpperWordStyle,
-        upper ? allUpperWordStyle : allLowerWordStyle,
+        upper ? s => capitalize(acronymsStyle(s)) : acronymsStyle,
```

Comment:
The lower option should always be `allLowerWordStyle`.  Otherwise, if you selected Pascal, you'd get `fooData` for `foo_data`, but `JSONData` for `json_data`.

Reply:
Pushed :)